pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -704,6 +704,11 @@ self: super: builtins.intersectAttrs super {
     librarySystemDepends = drv.librarySystemDepends or [] ++ [ pkgs.cyrus_sasl.dev ];
   }) super.LDAP);
 
+  # lsp 2.1.0.0 requires lsp-types >= 2.0.1 & < 2.1
+  lsp_2_1_0_0 = super.lsp_2_1_0_0.overrideScope (hfinal: hprev: {
+    lsp-types = hfinal.lsp-types_2_0_1_0;
+  });
```

Comment:
We generally ask for Haskell-related PRs to be based on the `haskell-updates` branch.

In the `haskell-updates` branch, `haskellPackages.lsp` is already 2.1.0.0, so I don't think this PR is necessary?  (It may become necessary when a new version of `lsp` is released, and we have some package/user that specifically wants to use `lsp-2.1.0.0`, but in general it likely won't be necessary.)